common/djangoapps/third_party_auth/samlproviderdata/views.py

```diff
@@ -0,0 +1,54 @@
+"""
+    Viewset for auth/saml/v0/samlproviderdata
+"""
+
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from third_party_auth.samlutils.utils import fetch_enterprise_customer_by_id
+
+from ..models import SAMLProviderConfig, SAMLProviderData
+from .serializers import SAMLProviderDataSerializer
+
+
+class SAMLProviderDataMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderDataSerializer
+
+
+class SAMLProviderDataViewSet(PermissionRequiredMixin, SAMLProviderDataMixin, viewsets.ModelViewSet):
```

Comment:
Same idea here about inheriting from more specific mixins.

Reply:
I will certainly revisit this in light of what comes out of work with Talia going forward, thanks